Xamarin.Forms.Platform.Android/AppCompat/CheckBoxRenderer.cs

```diff
@@ -0,0 +1,258 @@
+			SetOnCheckedChangeListener(this);
+
+			Tag = this;
+		}
+
+		protected override void Dispose(bool disposing)
+		{
+			if (_disposed)
+				return;
+
+			_disposed = true;
+
+			if (disposing)
+			{
+				_tracker?.Dispose();
+				_tracker = null;
+
+
+				if (CheckBox != null)
+				{
+					CheckBox.PropertyChanged -= OnElementPropertyChanged;
+
+					if (Android.Platform.GetRenderer(CheckBox) == this)
+					{
+						CheckBox.ClearValue(Android.Platform.RendererProperty);
+					}
+
+					CheckBox = null;
+				}
+			}
+
+			base.Dispose(disposing);
+		}
+
+		public override void Invalidate()
+		{
+			if (_skipInvalidate)
+			{
+				_skipInvalidate = false;
+				return;
+			}
+
+			base.Invalidate();
+		}
+
+		Size MinimumSize()
+		{
+			return new Size();
+		}
+
+		SizeRequest IVisualElementRenderer.GetDesiredSize(int widthConstraint, int heightConstraint)
+		{
+			if (_disposed)
+			{
+				return new SizeRequest();
+			}
+			Measure(widthConstraint, heightConstraint);
+			return new SizeRequest(new Size(MeasuredWidth, MeasuredHeight), MinimumSize());
+		}
+
+		void IVisualElementRenderer.SetElement(VisualElement element)
+		{
+
+			if (element == null)
+			{
+				throw new ArgumentNullException(nameof(element));
+			}
+
+			if (!(element is CheckBox checkBox))
+			{
+				throw new ArgumentException("Element is not of type " + typeof(CheckBox), nameof(element));
+			}
+
+			CheckBox oldElement = CheckBox;
+			CheckBox = checkBox;
+
+			Performance.Start(out string reference);
+
+			if (oldElement != null)
+			{
+				oldElement.PropertyChanged -= OnElementPropertyChanged;
+			}
+
+			element.PropertyChanged += OnElementPropertyChanged;
+
+			if (_tracker == null)
+			{
+				_tracker = new VisualElementTracker(this);
+
+			}
+
+			if (_visualElementRenderer == null)
+			{
+				_visualElementRenderer = new VisualElementRenderer(this);
+			}
+
+			Performance.Stop(reference);
+			this.EnsureId();
+						
+			UpdateOnColor();
+
+			ElementChanged?.Invoke(this, new VisualElementChangedEventArgs(oldElement, CheckBox));
+			CheckBox?.SendViewInitialized(Control);
+		}
+
+		// CheckBox related
+		void OnElementPropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			if (e.PropertyName == CheckBox.TintColorProperty.PropertyName)
+			{
+				UpdateOnColor();
+			}
+
+			ElementPropertyChanged?.Invoke(this, e);
+		}
+
+		void IOnCheckedChangeListener.OnCheckedChanged(CompoundButton buttonView, bool isChecked)
+		{
+			((IElementController)CheckBox).SetValueFromRenderer(CheckBox.IsCheckedProperty, isChecked);
+		}
+
+		void UpdateOnColor()
+		{		
+
+			if (CheckBox == null || Control == null)
+				return;
+
+
+			var mode = PorterDuff.Mode.SrcIn;
+
+			var stateChecked = global::Android.Resource.Attribute.StateChecked;
+			var stateEnabled = global::Android.Resource.Attribute.StateEnabled;
+			var statePressed = global::Android.Resource.Attribute.StatePressed;
+
+			//Need to find a way to get this color out of Android somehow.
```

Comment:
Is this comment still accurate? Is there a color we need to pull from the theme here?